Frameworks/UIKit/UIAlertView.mm

```diff
@@ -1,641 +1,671 @@
+            [alertPriv->_buttons[0]._buttonView sendControlEventsOnBack:UIControlEventTouchUpInside];
+        }
+    }
+
+    curHeight += 24.0f;
+
+    CGRect frame;
+    frame.origin.x = fullScreen.size.width / 2.0f - boxWidth / 2.0f;
+    frame.origin.y = fullScreen.size.height / 2.0f - curHeight / 2.0f + fullScreen.origin.x;
+    frame.size.width = boxWidth;
+    frame.size.height = curHeight;
+    [self initWithFrame:frame];
+
+    // Make sure we're not rotated even if WOCDisplayMode rotates the presentation surface
+    [self _handleRotation];
+
+    id image;
+    image = [[UIImage imageNamed:@"/img/alert-background@2x.png"] stretchableImageWithLeftCapWidth:25 topCapHeight:30];
+    UIImageSetLayerContents([self layer], image);
+
+    if (alertPriv->_messageLabel) {
+        [self addSubview:(id)alertPriv->_messageLabel];
+    }
+    if (alertPriv->_titleLabel) {
+        [self addSubview:(id)alertPriv->_titleLabel];
+    }
+
+    if (alertPriv->_cancelView) {
+        [self addSubview:(id)alertPriv->_cancelView];
+    }
+    for (int i = 0; i < alertPriv->_numButtons; i++) {
+        [self addSubview:(id)alertPriv->_buttons[i]._buttonView];
+    }
+
+    if ([alertPriv->_delegate respondsToSelector:@selector(willPresentAlertView:)]) {
+        [alertPriv->_delegate willPresentAlertView:self];
+    }
+
+    alertPriv->_darkView = [[UIView alloc] initWithFrame:fullScreen];
+    [self setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    [alertPriv->_darkView setBackgroundColor:[UIColor colorWithRed:1.0f green:1.0f blue:1.0f alpha:0.6f]];
+    [alertPriv->_darkView setAlpha:0.6f];
+    [alertPriv->_darkView setAutoresizingMask:UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth];
+    [self setAutoresizingMask:UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin |
+                              UIViewAutoresizingFlexibleBottomMargin | UIViewAutoresizingFlexibleTopMargin];
+    [self setAlpha:0.0f];
+    [popupWindow addSubview:alertPriv->_darkView];
+    [popupWindow addSubview:self];
+
+    [UIView beginAnimations:@"ShowAlert" context:nil];
+    [UIView setAnimationDuration:0.25f];
+    [alertPriv->_darkView setAlpha:1.0f];
+    [self setAlpha:1.0f];
+    [UIView commitAnimations];
+    alertPriv->_isVisible = true;
+
+    if ([alertPriv->_delegate respondsToSelector:@selector(didPresentAlertView:)]) {
+        [alertPriv->_delegate didPresentAlertView:self];
+    }
+}
+
+/**
+ @Status Interoperable
+*/
+- (void)dealloc {
+    for (int i = 0; i < alertPriv->_numButtons; i++) {
+        alertPriv->_buttons[i]._buttonView = nil;
+        alertPriv->_buttons[i]._buttonText = nil;
+    }
+    alertPriv->_title = nil;
+    alertPriv->_titleLabel = nil;
+    alertPriv->_message = nil;
+    alertPriv->_messageLabel = nil;
+    alertPriv->_cancelText = nil;
+    alertPriv->_cancelView = nil;
+
+    delete alertPriv;
+
+    [super dealloc];
+}

```

Comment:
Doh - nice catch.  Reinstall of git seems to have stomped my settings.

---
In reply to: [113998959](https://github.com/Microsoft/WinObjC/pull/2652#discussion_r113998959) [](ancestors = 113998959,113998799)